OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageAction.java

```diff
@@ -89,6 +99,15 @@
 
         if (json.has(TAGS))
             tags = new OSInAppMessageTag(json.getJSONObject(TAGS));
+
+        if (json.has(PROMPTS)) {
+            JSONArray promptsJsonArray = json.getJSONArray(PROMPTS);
```

Comment:
Create a class that handles this and use the constructor for it here and replace this with a one liner

Reply:
Uhm I don't believe that the handling from a JSONArray to a List needs a class, I will put it in a method though, to avoid that much constructions on the OSINAppMessageAction constructor

Reply:
I just wanted to it to look similar to `tags `